msal_extensions/token_cache.py

```diff
@@ -160,15 +156,14 @@ def __init__(self,
             lock_location=lock_location)
         self._dp_agent = WindowsDataProtectionAgent(entropy=entropy)
 
-    def _write(self):
+    def _write(self, contents):
         with open(self._cache_location, 'wb') as handle:
-            handle.write(self._dp_agent.protect(self.serialize()))
+            handle.write(self._dp_agent.protect(contents))
 
     def _read(self):
         with open(self._cache_location, 'rb') as handle:
```

Comment:
Again, by not inheriting the base method, we lost the customized exception handling logic here.

A different open mode seems not be a reason. In fact, Travis CI is also complaining the "rs" mode in the base class. :-)

Reply:
Fixed the `"rs"` thing in 9c59a72e9e737a711d0a17f64bb39225235d520a and added a test to get coverage of 9f459fb74fc5c2f5dd0952cf6b41e45368802833